pkgs/development/compilers/llvm/common/clang/default.nix

```diff
@@ -81,13 +81,17 @@ let
       # Make sure clang passes the correct location of libLTO to ld64
       substituteInPlace lib/Driver/ToolChains/Darwin.cpp \
         --replace-fail 'StringRef P = llvm::sys::path::parent_path(D.Dir);' 'StringRef P = "${lib.getLib libllvm}";'
-    '' + (if lib.versionOlder release_version "13" then ''
-      sed -i -e 's/DriverArgs.hasArg(options::OPT_nostdlibinc)/true/' \
-             -e 's/Args.hasArg(options::OPT_nostdlibinc)/true/' \
-             lib/Driver/ToolChains/*.cpp
-    '' else ''
-      (cd tools && ln -s ../../clang-tools-extra extra)
-    '') + lib.optionalString stdenv.hostPlatform.isMusl ''
+    '' + (
+      # See the comment on the `add-nostdlibinc-flag.patch` patch in
+      # `../default.nix` for why we skip Darwin here.
+      if lib.versionOlder release_version "13" && !stdenv.hostPlatform.isDarwin then ''
+        sed -i -e 's/DriverArgs.hasArg(options::OPT_nostdlibinc)/true/' \
+               -e 's/Args.hasArg(options::OPT_nostdlibinc)/true/' \
+               lib/Driver/ToolChains/*.cpp
+      '' else ''
+        (cd tools && ln -s ../../clang-tools-extra extra)
+      ''
+    ) + lib.optionalString stdenv.hostPlatform.isMusl ''
```

Comment:
this is removed in https://github.com/NixOS/nixpkgs/pull/348568

Reply:
Yes, https://github.com/NixOS/nixpkgs/pull/348568/commits/87086972d65c6a89b5a43182d6e845a80f807bce.

Reply:
It’s replaced with a patch, but I think the logic of not using it on Darwin would remain in that case.